packaging/requirements.py

```diff
@@ -92,9 +92,7 @@ def __init__(self, requirement_string):
         try:
             req = REQUIREMENT.parseString(requirement_string)
         except ParseException as e:
-            raise InvalidRequirement(
-                "Invalid requirement, parse error at \"{0!r}\"".format(
-                    requirement_string[e.loc:e.loc + 8]))
+            raise InvalidRequirement("Parse error: {0!r}".format(e))
```

Comment:
This changes the exception from:
```
Invalid requirement, parse error at "'42'"
```
to:
```
Parse error: Expected stringEnd (at char 5), (line:1, col:6)
```



It's not really clear to me how this improves the situation in https://github.com/pypa/pip/pull/5170? Is the idea that `pip` will now parse this new message and display something more meaningful to the user? I'm not sure that the new message is actually more user-friendly than the old one.